### PR TITLE
Net 447 (SLP-03) Adjust ticket block check

### PR DESCRIPTION
**Pull request by JCSanPedro**
## Motivation

The audit noted that it was possible for a ticket to be redeemed with a generation block greater than the current epoch's end block.

This PR prevents that by additionally checking if the ticket generation block is less than or equal to than the current block.

## Summary of Changes

- Add additional check for ticket generation block
- Remove require check that the ticket generation block is within the current epoch in the `redeem` function. This is because we also perform the same check in `calculateWinningProbability`, so it is unnecessary to do here as well.

### CODE REVIEW

**Review by JoshD641 — CHANGES_REQUESTED** [review]
The code change LGTM. At first I thought it was odd to remove the check from `redeem` rather than from `calculateWinningProbability`, and spread out the checks unnecessarily, but it does makes sense in the context of the ticket win probability decay to do it inside `calculateWinningProbability`.

We should probably leave the old test case in place, rather than editing it, to make sure that the require inside `calculateWinningProbability` is working too.

**Comment by JCSanPedro** [comment]
> We should probably leave the old test case in place, rather than editing it, to make sure that the require inside calculateWinningProbability is working too

There is another test case that already exists for that function specifically https://github.com/dn3010/sylo-ethereum-contracts/blob/master/test/payments/ticketing.ts#L414-L435

**Review by JoshD641 — APPROVED** [review]

File: contracts/Payments/Ticketing.sol [hunk 1]
```diff
@@ -225,9 +225,8 @@ contract SyloTicketing is Initializable, OwnableUpgradeable {
         EpochsManager.Epoch memory epoch = _epochsManager.getEpoch(ticket.epochId);
         require(epoch.startBlock > 0, "Ticket's associated epoch does not exist");
```
Comment (teinnt) [inline contracts/Payments/Ticketing.sol:226]: Do we want to remove the check here, because this will be checked again in `calculateWinningProbability`?

Comment (JCSanPedro) [reply contracts/Payments/Ticketing.sol:226]: Makes sense

**Review by teinnt — APPROVED** [review]

**Pull request merged**